### DISCUSSION TITLE
Unable to follow Getting Started Steps with GCP

### DISCUSSION THREAD

**Discussion by prahaladd** [edgelesssys/constellation#2956] [Q&A]
Hello,

I am attempting to get the community edition of constellation running on my GCP cluster.  I am following the instructions present on the page [First Steps With Constellation](https://docs.edgeless.systems/constellation/getting-started/first-steps)

So, I am able to create the configurations required for GCP as well as the IAM configs. However, running `constellation apply -` always times out. 
Please note that I am attempting to bring up a constellation cluster in `asia-south-1a` region of GCP since I am based out of India.
As can be seen within the shell excerpt, the context deadline has exceeded. This has occurred repeatedly and I can consistently reproduce the issue.
Additionally, while the message below says that `Cloud infrastructure created successfully`, I really do not see any of the VMs having been created under the `Compute Engine -> VM Instances` link in the GCP console. So this command is probably printing out wrong/premature status? Or does it only check for the creation of certain terraform scripts/files?

```shell
prahaladd@tranquility:~/Projects/edgeless/constellation$ constellation apply -y
Using community license.
For details, see https://docs.edgeless.systems/constellation/overview/license
Checking for infrastructure changes  
The following Constellation cluster will be created:
  3 control-plane nodes of type n2d-standard-4 will be created.
  1 worker node of type n2d-standard-4 will be created.
Creating  
Cloud infrastructure created successfully.
Your Constellation master secret was successfully written to "constellation-mastersecret.json"
Connecting   
Error: doing init call: context deadline exceeded
```
Refer attached image for empty VM list in the GCP console
![Screenshot from 2024-02-28 23-58-33](https://github.com/edgelesssys/constellation/assets/441921/6671927c-fec4-400a-9aae-feb65da6151a)

**Comment by malt3** [comment]
Hi! Thanks for reaching out. Can you check if the VMs are part of a different GCP project you are logged in to?
You can check via `gcloud config configurations list`.

If you find the VMs, can you check if they are healthy and look into the serial log to check for boot issues?

**Reply by prahaladd** [reply]
Hello @malt3 - thank you so much for these insights. I did check my script and I realized that my `gcloud` command was using an application default belonging to another account itself. I have now corrected and things go ahead but fail with quota limits. I am on GCP free tier at the moment and it seems like constellation setup takes up resources that do not fit within the GCP free tier quota.
We can update the document to ensure that this is mentioned? Or should there be a way to work with reduced resource counts that are within the quota so that early users can experiment quickly for an evaluation?

Also IMHO there should be a command in constellation to completely clean slate the account of all the artifacts such as networks, IP addresses, TCP proxies created by the install script to be deleted so that these do not add up to project quotas after their use is complete.  I can log this as a separate ticket for this.

**Reply by malt3** [reply]
I think it is generally expected for software deployed in the cloud to incur some resource cost. I don't believe we can offer a representative, nice to use preview of Constellation using only the GCP free tier.

There is a command to clean up a Constellation. Just enter `constellation terminate`.
